crates/biome_js_analyze/src/lint/nursery/no_evolving_any.rs

```diff
@@ -0,0 +1,115 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::{AnyJsExpression, JsFileSource, JsVariableDeclaration, JsVariableDeclarator};
+
+declare_rule! {
+    /// Disallow variables from evolving into `any` type through reassignments.
+    ///
+    /// In TypeScript, variables without explicit type annotations can evolve their types based on subsequent assignments.
+    /// This behavior can accidentally lead to variables with an `any` type, weakening type safety.
+    /// Just like the `any` type, evolved `any` types disable many type checking rules and should be avoided to maintain strong type safety.
+    /// This rule prevents such cases by ensuring variables do not evolve into `any` type, encouraging explicit type annotations and controlled type evolutions.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```ts,expect_diagnostic
+    /// let a;
+    /// const b = [];
+    /// let c = null;
```

Comment:
I am not sure if you read the contribution guide, but when creating a snippet that should emit diagnostics, you should make sure that *each snippet* emits **ONLY one diagnostic**.

https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md#document-the-rule

> When adding invalid snippets in the ### Invalid section, you must use the expect_diagnostic code block property. We use this property to generate a diagnostic and attach it to the snippet. A snippet must emit only ONE diagnostic.

This means that you have to break down this snippet in THREE code blocks:

`````md
```ts,expect_diagnostic
let a;
```

```ts,expect_diagnostic
const b = [];
```
`````

Reply:
I sincerely apologize for overlooking that rule. 🙏 I will make the necessary corrections promptly.

Reply:
No problem! That's what a review is for :) catch possible bugs!